rest_framework/fields.py

```diff
@@ -1406,6 +1411,11 @@ def to_internal_value(self, data):
     def to_representation(self, value):
         if value in ('', None):
             return value
+
+        if isinstance(value, (IntegerChoices, TextChoices)) and str(value) != \
```

Comment:
I have one concerns/question. should we consider both Int and text value as string?

Reply:
I've typecasted both the `__str__` representation of `value` and `value.value` for the check here.